source/georef/street_network.cpp

```diff
@@ -412,28 +450,37 @@ PathFinder::find_nearest_stop_points(const navitia::time_duration& radius,
                 // if the radius is still ok with sqrt(2) factor
                 auto sp_idx = routing::SpIdx(element.first);
                 if (duration < radius && distance_to_entry_point.count(sp_idx) == 0) {
-                    result[sp_idx] = duration;
-                    distance_to_entry_point[sp_idx] = duration;
+
+                    // free radius exclusion (optional)
+                    if (free_radius > 0 && !excluded_elements.empty()) {
```

Comment:
It might be easier to read if this case was handled by `exclude_sp_with_free_radius_filter()`. But that's only an opinion on with you don't have to agree.